test/functional/eval/environ_spec.lua

```diff
@@ -17,3 +22,63 @@ describe('environment variables', function()
     eq(0, exists('$DOES_NOT_EXIST'))
   end)
 end)
+
+describe('empty $HOME', function()
+  local original_home = ''
+
+  -- save $HOME before each test
+  before_each(function()
+    original_home = os.getenv('HOME')
+  end)
+
+  -- recover $HOME after each test
+  after_each(function()
+    if original_home ~= nil then
+      setenv('HOME', original_home)
+    end
```

Comment:
when would `original_home` be `nil`?

Reply:
I believe when the environment variable HOME does not exist or unset, os.getenv() returns a nil. I originally didn't have this guard and got a "Vim:E119: Not enough arguments for function: setenv" error.